src/client/components/table-of-contents/table-of-contents.vue

```diff
@@ -14,5 +15,78 @@
 <script>
 export default {
   props: ['items'],
+  data () {
+    return {
+      currentSection: this.items[0].slug
+    }
+  },
+  mounted () {
+    window.addEventListener('scroll', this.getClosestSection)
+  },
+  
+  beforeDestroy () {
+    window.removeEventListener('scroll', this.getClosestSection)
+  },
+
+  methods: {
+    getClosestSection () {
+      let sectionsLenght = this.items.length
+
+      for (var i = 0; i < sectionsLenght; i++) {
+        let myElement = document.getElementById(this.items[i].slug)
+
+        if (this.isBelowScroll(myElement)) {
+          this.currentSection = myElement.id
+        }
+      }
+    },
+
+    isBelowScroll (el) {
+      let position = el.getBoundingClientRect()
+      
+      return position.top < 200 && ((position.top + position.height) > 0)
+    }
+  }
 }
 </script>
+
+<style>
+.table-of-contents {
+  display: block;
+}
+
+.table-of-contents__inline-list {
+  padding: 0;
+  text-align: left;
+}
+
+.table-of-contents__list-item {
+  list-style-type: none;
+  text-transform: uppercase;
+}
+
+.table-of-contents__list-item:not(:first-child) .table-of-contents__link-item {
+  padding-top: var(--spacing-default);
+}
+
+.table-of-contents__link-item {
+  display: block;
+  padding-bottom: var(--spacing-default);
+  color: var(--text-color);
+  background: linear-gradient(to right, var(--action-color), var(--action-color));
+  background-size: 0% 3px;
+  background-position: bottom left;
+  background-repeat: no-repeat;
+  -webkit-transition: 1s;
```

Comment:
You shouldn't need to prefix manually. If this is an issue, we need to change the rules in our config.